### PR TITLE
Update for DynUNet's changes and fix get_feature_maps issue

### LINKED ISSUE #106: DynUNet needs updating for deep_supervision
**Describe the bug**
Following [this PR](https://github.com/Project-MONAI/MONAI/pull/1393) to DynUNet, [the corresponding notebook](https://github.com/Project-MONAI/tutorials/blob/master/modules/dynunet_tutorial.ipynb) needs updating.

I tried simply removing `deep_supervision = True` from the constructor, but this wasn't enough.

The subsequent error is attached here: [dynunet.txt](https://github.com/Project-MONAI/tutorials/files/5808380/dynunet.txt).

@Nic-Ma I think you might best know what needs to be changed here.

### LINKED ISSUE #128: Error when running 'DynUNet' tutorial:  object has no attribute 'get_feature_maps'
**Describe the bug**

I tried to run this tutorial, and I got this error message:
"torch.nn.modules.module.ModuleAttributeError: 'DynUNet' object has no attribute 'get_feature_maps'"
This error showed when runing the docker images "projectmonai/monai:0.4.0" and 2projectmonai/monai:latest".

It seems to me that the DynUNet public interface "get_feature_maps method" was deleted and updated with a concatenated tensor. However, the tutorial still not updated.

Please only create issues here relating to the functioning of the tutorials. For all other questions/problems/bugs, please refer to MONAI's [main repository](https://github.com/Project-MONAI/MONAI). For MONAI bugs (not just a problem in the notebook), please create an [issue](https://github.com/Project-MONAI/MONAI/issues). For questions, please use the [Discussions tab](https://github.com/Project-MONAI/MONAI/discussions).

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes #106 and #128 .

**Description**
Due to the forward function's changes of DynUNet, this commit
update the corresponding places, as well as the loss calculation
for trainer.

In addition, the DiceCEloss has been implemented in MONAI, thus the
self-designed loss function part has also been updated.

Signed-off-by: Yiheng Wang <vennw@nvidia.com>

### CODE REVIEW

File: modules/dynunet_tutorial.ipynb [hunk 1]
```diff
@@ -616,30 +577,26 @@
...
     "\n",
     "        def _compute_loss(preds, label):\n",
-    "            labels = [label] + [\n",
-    "                interpolate(label, pred.shape[2:]) for pred in preds[1:]\n",
-    "            ]\n",
+    "            preds = torch.unbind(preds, dim=1)\n",
     "            return sum(\n",
-    "                0.5 ** i * self.loss_function(p, l)\n",
-    "                for i, (p, l) in enumerate(zip(preds, labels))\n",
+    "                [\n",
```
Comment (rijobro) [inline modules/dynunet_tutorial.ipynb:582]: Small point, but list comprehension is unnecessary here as `sum` can take a generator as argument. i.e., you could delete the `[` and `]` and outdent the rest.

**Comment by rijobro** [comment]
@yiheng-wang-nv thanks! Do you think you could add some comments explaining the shape of the tensor returned by `forward`? I'm guessing it's the `torch.unbind` that removes the feature maps? Could you add a comment to that affect, and mention that it's only necessary when `deep_supervision==True`?

**Comment by rijobro** [comment]
fixes #128.

**Comment by coranium** [comment]
Hi, somehow am getting labels of shape (2, 1,  40, 56, 40) for taskid = 4.

This causes an error with non-matching dimensions between predictions (4) and targets(5).

I'm guessing onehot_y should be = False and need to squeeze labels. Unless some of the decathlon tasks do not have one-hot-labels

**Comment by rijobro** [comment]
Hi @coranium this seems like a new issue to me. Could you create a new Discussion, preferably with a minimal working example.

**Comment by yiheng-wang-nv** [comment]
Hi all, I opened a new draft PR, see #131 , since the current tutorial version is too old, which may have many issues, I rewrite a new python file based pipeline, and have tested on all decathlon tasks locally. Since my local version does not have enough doc strings, and used `DataParallel` rather than `DistributedDataParallel` for multi-gpu training, to meet the latest MONAI version, the PR is still in processing.
Please take a look at #131 for further updates, thanks!

**Comment by wyli** [comment]
Hi @yiheng-wang-nv do you have an ETA for the new pipeline? if it'll take some time, is it possible to provide a quick fix for  https://github.com/Project-MONAI/tutorials/issues/128 first?

**Comment by yiheng-wang-nv** [comment]
> Hi @yiheng-wang-nv do you have an ETA for the new pipeline? if it'll take some time, is it possible to provide a quick fix for #128 first?

I see, let me reopen this PR, and create another one to distinguish the new pipeline.

**Comment by yiheng-wang-nv** [comment]
> @yiheng-wang-nv thanks! Do you think you could add some comments explaining the shape of the tensor returned by `forward`? I'm guessing it's the `torch.unbind` that removes the feature maps? Could you add a comment to that affect, and mention that it's only necessary when `deep_supervision==True`?

Hi @rijobro , I've added the comments, could you please help to review it again? As for the new pipeline I mentioned above, I'll open another branch and submit a PR after finishing it.

**Review by wyli — APPROVED** [review]
thanks it looks good to me

**Comment by rijobro** [comment]
@yiheng-wang-nv added comments look good, thanks!

**Pull request merged**